logsmanagement/parser.c

```diff
@@ -0,0 +1,1622 @@
+            debug(D_LOGS_MANAG, "Extracted RESP_CODE:%d", log_line_parsed->resp_code);
+            #endif
+
+            goto next_item;
+        }
+
+        if(fields_format[i] == RESP_SIZE){
+            /* TODO: Differentiate between '-' or 0 and an invalid response size. 
+             * right now, all these will set resp_size == 0 */
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Item %d (type: RESP_SIZE):%.*s", i, (int)field_size, field);
+            #endif
+
+            char resp_size_d[REQ_RESP_SIZE_MAX_LEN];
+            snprintf( resp_size_d, 
+                      field_size > (REQ_RESP_SIZE_MAX_LEN - 1) ? REQ_RESP_SIZE_MAX_LEN : field_size + 1, 
+                      "%.*s", (int)field_size, field);
+
+            if(field[0] ==  '-' && field_size == 1) { 
+                log_line_parsed->resp_size = 0; // Response size can be '-' 
+            }
+            else if(likely(str2int(&log_line_parsed->resp_size, resp_size_d, 10) == STR2XX_SUCCESS)){
+                if(verify){
+                    if(unlikely(log_line_parsed->resp_size < 0)){
+                        #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                        error("RESP_SIZE is invalid (<0)");
+                        #endif
+                        log_line_parsed->resp_size = 0;
+                        log_line_parsed->parsing_errors++;
+                    }
+                }
+            }
+            else {
+                #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+                error("Error while extracting RESP_SIZE from string");
+                #endif
+                log_line_parsed->resp_size = 0;
+                log_line_parsed->parsing_errors++;
+            }
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Extracted RESP_SIZE:%d", log_line_parsed->resp_size);
+            #endif
+
+            goto next_item;
+        }
+
+        if(fields_format[i] == UPS_RESP_TIME){
+            #if ENABLE_PARSE_WEB_LOG_LINE_DEBUG
+            debug(D_LOGS_MANAG, "Item %d (type: UPS_RESP_TIME):%.*s", i, (int)field_size, field);
+            #endif
+
+            if(field[0] ==  '-' && field_size == 1) { 
+                log_line_parsed->ups_resp_time = 0;
+                log_line_parsed->parsing_errors++;
+                goto next_item;
+            }
+
+            /* Times of several responses are separated by commas and colons. Following the 
+             * Go parser implementation, where only the first one is kept, the others are 
+             * discarded. Also, there must be no space in between them. Needs testing... */
+            char *pch = memchr(field, ',', field_size);
+            if(pch) field_size = pch - field;
+
+            float f = 0;
+
+            char ups_resp_time_d[UPS_RESP_TIME_MAX_LEN];
+            snprintf( ups_resp_time_d, 
+                      field_size > (UPS_RESP_TIME_MAX_LEN - 1) ? UPS_RESP_TIME_MAX_LEN : field_size + 1, 
```

Comment:
please, take a look in all `snprintf`, convert them to `snprintfz` and adjust limits. I won't mark the issue again after this message.